.github/workflows/build-binaries.yml

```diff
@@ -21,42 +21,58 @@ jobs:
   build-bootstrapper:
     runs-on: ubuntu-latest
     steps:
+      - name: Checkout on PR
+        if: github.event_name == 'pull_request'
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
I _think_ you don't actually need to create two conditional steps here. I looked a bit around on GitHub and found repos that also run on both pull_request and push and specify `ref: ${{ github.event.pull_request.head.sha }}`  for both.

I think it makes sense and can work, since in case of a push, the ref would likely be an empty string, causing the action to use the default behavior (ref = SHA that triggered the action) rather than failing.

Can you please try this? You could keep temporarily remove the pull_request trigger and change it to trigger on a push to your branch while keeping the same
```yaml
       with:
          ref: ${{ github.event.pull_request.head.sha }}
```
variable passthrough.